### PR TITLE
Fix warning when asserting a pattern that always match

### PR DESCRIPTION
**Pull request by tsubery**
Assertion macro generated a case statement that can have patterns that
cannot match. For example:

```Elixir
test "warning" do
  assert a = %{a: 1}[:a]
  a #don't complain about unused var
end
```

Would generate the following output
```Bash
warning: this clause cannot match because a previous clause at line 7
always matches
```

### CODE REVIEW

File: lib/ex_unit/lib/ex_unit/assertions.ex [hunk 1]
```diff
@@ -138,6 +136,12 @@ defmodule ExUnit.Assertions do
               message: "match (=) failed" <>
                        ExUnit.Assertions.__pins__(unquote(pins))
         end
+      end)
+
+    quote do
+      right = unquote(right)
+      expr  = unquote(code)
```
Comment (lexmag) [inline lib/ex_unit/lib/ex_unit/assertions.ex]: Let's get rig of `=` alignment.

File: lib/ex_unit/lib/ex_unit/assertions.ex [hunk 2]
```diff
@@ -123,10 +123,8 @@ defmodule ExUnit.Assertions do
         end
       end)
 
-    quote do
-      right = unquote(right)
-      expr  = unquote(code)
-      unquote(vars) =
+    match_case =
```
Comment (lexmag) [inline lib/ex_unit/lib/ex_unit/assertions.ex]: I think `case_expr` would read better, what do you think?

File: lib/ex_unit/lib/ex_unit/assertions.ex (hunk 1 above)
Comment (tsubery) [inline lib/ex_unit/lib/ex_unit/assertions.ex]: sure

File: lib/ex_unit/lib/ex_unit/assertions.ex (hunk 2 above)
Comment (tsubery) [inline lib/ex_unit/lib/ex_unit/assertions.ex]: In general I rather name things for their intention over what they happen to be. In this case the intention is to do pattern matching and using "case" could be regarded as an is implementation detail. Don't feel strongly about it though, If the variable assignment is only two lines away naming is less important.

Comment (lexmag) [inline lib/ex_unit/lib/ex_unit/assertions.ex]: Agree, `match_case` just for a seconds looked confusing to me, maybe `match_expr` would be even better?

Comment (tsubery) [inline lib/ex_unit/lib/ex_unit/assertions.ex]: ```match_expr``` sounds good

**Comment by josevalim** [comment]
:heart: :green_heart: :blue_heart: :yellow_heart: :purple_heart:

**Pull request merged**